glean-core/ios/Glean/Debug/GleanDebugTools.swift

```diff
@@ -0,0 +1,141 @@
+    /// - `sendPing`: This command expects a string name of a ping to force immediate collection and submission of.
+    ///
+    /// The structure of the custom URL uses the following format:
+    ///
+    /// `<protocol>://glean?<command 1>=<paramter 1>&<command 2>=<parameter 2> ...`
+    ///
+    /// Where:
+    ///
+    /// - `<protocol>://` is the "Url Scheme" that has been added for the app and doesn't matter to Glean.
+    /// - `glean` is required for the Glean SDK to recognize the command is meant for it to process.
+    /// - `?` indicating the beginning of the query.
+    /// - `<command>=<parameter>` are instances of the commands listed above  separated by `&`.
+    ///
+    /// There are a few things to consider when creating the custom URL:
+    ///
+    /// - Invalid commands will trigger an error and be ignored.
+    /// - Not all commands are requred to be encoded in the URL, you can mix and match the commands that you need.
+    /// - Special characters should be properly URL encoded and escaped since this needs to represent a valid URL.
+    static func handleCustomUrl(url: URL) {
+        guard url.host?.removingPercentEncoding == "glean" else {
+            return
+        }
+
+        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: true),
+            let params = components.queryItems else {
+            logger.error("Error parsing query parameters, aborting Glean debugging tools")
+            return
+        }
+
+        if let parsedCommands = processCustomUrlQuery(urlQueryItems: params) {
+            if let tag = parsedCommands.pingTag {
+                Glean.shared.configuration?.pingTag = tag
+                logger.debug("Pings tagged with: \(tag)")
+            }
+
+            if let logPings = parsedCommands.logPings {
+                Glean.shared.configuration?.logPings = logPings //
+                logger.debug("Log pings set to: \(logPings)")
+            }
+
+            if let pingName = parsedCommands.pingNameToSend {
+                Glean.shared.sendPingsByName(pingNames: [pingName])
+                logger.debug("Glean debug tools triggered ping: \(pingName)")
+            }
+        }
+    }
+
+    /// A simple struct to represent the commands parsed from the custom URL
+    private struct ParsedQueryCommands {
+        let pingTag: String?
+        let logPings: Bool?
+        let pingNameToSend: String?
+    }
+
+    /// Helper function to process parameters passed as a query to the custom URL processed by `handleCustomUrl`
+    ///
+    /// - parameters:
+    ///     * urlQueryItems: A `URLQueryItem` representing the commands passed to Glean
+    ///
+    /// - returns: A tuple containing the values of the parameters `pingTag`, `logPings`, `pingNamesToSend` or nil.
+    private static func processCustomUrlQuery(urlQueryItems: [URLQueryItem]) -> ParsedQueryCommands? {
```

Comment:
Yey! I do like this refactoring into its own function! :)
Much better then what I had in mind doing that inline :D